src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt

```diff
@@ -14,18 +14,21 @@ import org.jmailen.gradle.kotlinter.id
 internal object AndroidSourceSetApplier : SourceSetApplier {
 
     override fun applyToAll(project: Project, action: SourceSetAction) {
-        val android = project.extensions.findByName("android")
-        (android as? BaseExtension)?.let {
-            it.sourceSets.all { sourceSet ->
-                val id = sourceSet.name.id
-                action(id, project.provider { getKotlinFiles(project, sourceSet) })
-            }
+        val android = project.extensions.findByName("android") as? BaseExtension ?: return
+        android.sourceSets.configureEach { sourceSet ->
+            val id = sourceSet.name.id
+            action(id, project.provider { getKotlinFiles(project, sourceSet) })
         }
     }
 
-    private fun getKotlinFiles(project: Project, sourceSet: AndroidSourceSet) = sourceSet.java.srcDirs.map { dir ->
-        project.fileTree(dir) { it.include("**/*.kt") }
-    }.reduce { merged: FileTree, tree: ConfigurableFileTree ->
-        merged + tree
+    private fun getKotlinFiles(project: Project, sourceSet: AndroidSourceSet): FileTree? {
+        val javaSources = sourceSet.java.srcDirs
+        val kotlinSources = runCatching { (sourceSet.kotlin as? com.android.build.gradle.api.AndroidSourceDirectorySet)?.srcDirs }
+            .getOrNull()
+            .orEmpty()
```

Comment:
This is a _trick_ which I read as "if there is a speparate kotlin sourceset append it to java ones". I initially wanted to do a version check on AGP, but I couldn't find a stable api to do so. 
I once reported: https://issuetracker.google.com/issues/188859855 where I tried to let AGP maintainers know it's unclear what's the preferred way to obtain AGP version. 
The alternatives I considered are:
- add their internal artifact and use Version object
- continue using deprecated property until they remove it at some point
- do a reflective call to check if `kotlin` property exists

As a conclusion I'm proposing to do catch and ignore the exception if AGP 7.0 specific classes aren't available in runtime

Reply:
That seems like a reasonable approach all considered. This or that.